pkg/daemon/update.go

```diff
@@ -145,7 +153,7 @@ func (dn *Daemon) performPostConfigChangeAction(postConfigChangeActions []string
 	return nil
 }
 
-func (dn *Daemon) getAndUpdateConfigAndState(configName string) error {
+func (dn *MCUpdater) getAndUpdateConfigAndState(configName string) error {
```

Comment:
Use a different variable name for the pointer receiver since we're somewhat decoupling this, e.g.:

```golang
func (mcu *MCUpdater) getAndUpdateConfigAndState(configName string) error {
    state, err := mcu.getStateAndConfigs(configName)
    // ...
}
```